src/sentry/incidents/models.py

```diff
@@ -399,6 +406,52 @@ def clear_alert_rule_subscription_caches(cls, instance, **kwargs):
                 for sub_id in subscription_ids
             )
 
+    def conditionally_subscribe_project_to_alert_rules(
+        self,
+        project: Project,
+        activation_condition: AlertRuleActivationConditionType,
+        query_extra: str,
+        trigger: str,
+    ):
+        """
+        Subscribes a project to an alert rule given activation condition
+        """
+        try:
+            project_alert_rules: list[AlertRule] = self.filter(
+                project=project, monitor_type=AlertRuleMonitorType.ACTIVATED
+            )
+            created_subscriptions = []
+            if project_alert_rules.exists():
+                # If we have an AlertRule for the project
+                for alert_rule in project_alert_rules:
+                    if alert_rule.activation_conditions.filter(
+                        condition_type=activation_condition
+                    ).exists():
+                        logger.info(
+                            "Attempt subscribe project to activated alert rule",
+                            extra={
+                                "trigger": trigger,
+                                "query_extra": query_extra,
+                            },
+                        )
+                        created_subscriptions.append(
```

Comment:
`.extend(...)`

Reply:
nice good catch. that would've been annoying to debug 😅